src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -467,20 +467,20 @@ private void ParseNamespaceBody(ref SyntaxToken openBrace, ref NamespaceBodyBuil
                             }
                             else
                             {
-                                // incomplete members must be processed before we add any nodes to the body:
-                                ReduceIncompleteMembers(ref pendingIncompleteMembers, ref openBrace, ref body, ref initialBadNodes);
+                                parseUsingDirective(ref openBrace, ref body, ref initialBadNodes, ref seen, ref pendingIncompleteMembers);
+                            }
 
-                                var @using = this.ParseUsingDirective();
-                                if (seen > NamespaceParts.Usings)
-                                {
-                                    @using = this.AddError(@using, ErrorCode.ERR_UsingAfterElements);
-                                    this.AddSkippedNamespaceText(ref openBrace, ref body, ref initialBadNodes, @using);
-                                }
-                                else
-                                {
-                                    body.Usings.Add(@using);
-                                    seen = NamespaceParts.Usings;
-                                }
+                            reportUnexpectedToken = true;
+                            break;
+
+                        case SyntaxKind.IdentifierToken:
+                            if (this.CurrentToken.ContextualKind != SyntaxKind.GlobalKeyword || this.PeekToken(1).Kind != SyntaxKind.UsingKeyword)
```

Comment:
fwiw, i think tihs would be much clearer if hte if was flipped and the condition was just `if (CurrentToken == Global && PeekToken(1) == Using)`.

Reply:
> fwiw, i think tihs would be much clearer if hte if was flipped and the condition was just if (CurrentToken == Global && PeekToken(1) == Using).

I prefer to structure the code in this case section and in the previous section as close as possible.

---
In reply to: [581474923](https://github.com/dotnet/roslyn/pull/51426#discussion_r581474923) [](ancestors = 581474923)